src/lib/es2015.reflect.d.ts

```diff
@@ -1,15 +1,43 @@
 declare namespace Reflect {
-    function apply(target: Function, thisArgument: any, argumentsList: ArrayLike<any>): any;
-    function construct(target: Function, argumentsList: ArrayLike<any>, newTarget?: any): any;
+    /**
+     * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
+     * @param target The function to call.
+     * @param thisArgument The object to be used as the this object.
+     * @param argumentsList An array of argument values to be passed to the function.
+     */
+    function apply<T, A extends any[], R>(target: (this: T, ...args: A) => R, thisArgument: T, argumentsList: A): R;
+    function apply(target: (...args: any) => any, thisArgument: any, argumentsList: ArrayLike<any>): any;
+
+    /**
+     * Constructs the target with the elements of specified array as the arguments and the specified constructor as the `new.target` value.
+     * @param target The function to call.
+     * @param argumentsList An array of argument values to be passed to the function.
+     * @param newTarget The constructor to be used as the `new.target` object.
+     */
+    function construct<A extends any[], R>(target: new (...args: A) => R, argumentsList: A, newTarget?: new (...args: any) => any): R;
+    function construct(target: new (...args: any) => any, argumentsList: ArrayLike<any>, newTarget?: new (...args: any) => any): any;
+
     function defineProperty(target: object, propertyKey: PropertyKey, attributes: PropertyDescriptor): boolean;
     function deleteProperty(target: object, propertyKey: PropertyKey): boolean;
     function get(target: object, propertyKey: PropertyKey, receiver?: any): any;
     function getOwnPropertyDescriptor(target: object, propertyKey: PropertyKey): PropertyDescriptor | undefined;
-    function getPrototypeOf(target: object): object;
+
+    /**
+     * Returns the prototype of an object.
+     * @param o The object that references the prototype.
+     */
+    function getPrototypeOf(target: object): object | null;
     function has(target: object, propertyKey: PropertyKey): boolean;
     function isExtensible(target: object): boolean;
     function ownKeys(target: object): PropertyKey[];
     function preventExtensions(target: object): boolean;
     function set(target: object, propertyKey: PropertyKey, value: any, receiver?: any): boolean;

```

Comment:
`set` / `has` / `get` could benefit from overloads like the ones made for `apply` and `construct`, though; `get` in particular.